common/lib/common-utils/src/assert.ts

```diff
@@ -3,14 +3,37 @@
  * Licensed under the MIT License.
  */
 
+import { FluidError } from "./fluidError";
+
+export class AssertError extends FluidError {
+    constructor(
+        errorMessage: string,
+        readonly errorType: string,
+    ) {
+        super(errorMessage);
+    }
+}
+
 /**
  * A browser friendly version of the node assert library. Use this instead of the 'assert' package, which has a big
  * impact on bundle sizes.
  * @param condition - The condition that should be true, if the condition is false an error will be thrown.
  * @param message - The message to include in the error when the condition does not hold
  */
- export function assert(condition: boolean, message?: string): asserts condition {
+ export function simpleAssert(condition: boolean, message?: string): asserts condition {
```

Comment:
i assume we would deprecate this

Reply:
Yes

Reply:
Why not delete it? I.e. what's the scenario where it's needed?

Reply:
Just to make it more mechanical to consume the updated version of common-utils (find all build breaks and switch to simpleAssert). Then in separate PR(s) we can replace `simpleAssert` with `assert`.